diff --git a/libraries/AP_HAL/board/linux.h b/libraries/AP_HAL/board/linux.h
index 1bbdcddd5f..81c7f7f5ac 100644
--- a/libraries/AP_HAL/board/linux.h
+++ b/libraries/AP_HAL/board/linux.h
@@ -293,7 +293,8 @@
     //#define HAL_BARO_ALLOW_INIT_NO_BARO
     
     // Activate SUB Model Configuratopm
-    #define HAL_BOARD_SUBTYPE_LINUX_OBAL_V1_MPU_9250_SPI
+    //#define HAL_BOARD_SUBTYPE_LINUX_OBAL_V1_MPU_9250_SPI
+    #define HAL_BOARD_SUBTYPE_LINUX_OBAL_V1_GY_91_SPI
     
     // Define Notify
     #define OBAL_NOTIFY_LED
@@ -301,18 +302,27 @@
 
     // GY-91 SPI Connection
     #ifdef HAL_BOARD_SUBTYPE_LINUX_OBAL_V1_MPU_9250_SPI
+        #define HAL_BOARD_STORAGE_DIRECTORY "/home/pi/ardupilot"
         #define HAL_BOARD_LOG_DIRECTORY "/home/pi/ardupilot/logs"
         #define HAL_BOARD_TERRAIN_DIRECTORY "/home/pi/ardupilot/terrain"
-        #define HAL_BOARD_STORAGE_DIRECTORY "/home/pi/ardupilot"
         #define HAL_PARAM_DEFAULTS_PATH "/home/pi/ardupilot.parm"
 
         #define HAL_INS_PROBE_LIST PROBE_IMU_SPI(Invensense, "mpu9250", ROTATION_NONE)
         #define HAL_MAG_PROBE_LIST PROBE_MAG_IMU(AK8963, mpu9250, 0, ROTATION_NONE)
         #define HAL_BARO_PROBE_LIST PROBE_BARO_I2C(BMP085, 1, 0x77) 
-        //#define HAL_MAG_PROBE_LIST PROBE_MAG_I2C(QMC5883L, 1, 0x0d,true ,  ROTATION_NONE)
 
         #define HAL_PROBE_EXTERNAL_I2C_COMPASSES
     #endif
+    #ifdef HAL_BOARD_SUBTYPE_LINUX_OBAL_V1_GY_91_SPI
+        #define HAL_BOARD_STORAGE_DIRECTORY "/home/pi/ardupilot.gy91"
+        #define HAL_BOARD_LOG_DIRECTORY "/home/pi/ardupilot.gy91/logs"
<<<<<<< HEAD
+        #define HAL_PARAM_DEFAULTS_PATH "/home/pi/ardupilot.parm"
+    
=======
+        #define HAL_BOARD_TERRAIN_DIRECTORY "/home/pi/ardupilot/terrain"
+        #define HAL_PARAM_DEFAULTS_PATH "/home/pi/ardupilot.parm"
>>>>>>> 3c9d4f7221c10371119f50eb9c78f6e3425fe7db
+        #define HAL_INS_PROBE_LIST PROBE_IMU_SPI(Invensense, "mpu9250", ROTATION_NONE)
+        #define HAL_MAG_PROBE_LIST PROBE_MAG_IMU(AK8963, mpu9250, 0, ROTATION_NONE)
+        #define HAL_BARO_PROBE_LIST PROBE_BARO_SPI(BMP280, "bmp280")
+        #define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+    #endif
 
     
     #ifdef OBAL_NOTIFY_LED
diff --git a/libraries/AP_HAL_Linux/SPIDevice.cpp b/libraries/AP_HAL_Linux/SPIDevice.cpp
index 4d3e922561..463b2f6200 100644
--- a/libraries/AP_HAL_Linux/SPIDevice.cpp
+++ b/libraries/AP_HAL_Linux/SPIDevice.cpp
@@ -150,6 +150,12 @@ SPIDesc SPIDeviceManager::_device[] = {
 SPIDesc SPIDeviceManager::_device[] = {
     SPIDesc("mpu9250",    0, 0, SPI_MODE_3, 8, SPI_CS_KERNEL,  1*MHZ, 10*MHZ),
 };
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_OBAL_V1 && \
+    defined (HAL_BOARD_SUBTYPE_LINUX_OBAL_V1_GY_91_SPI)
+SPIDesc SPIDeviceManager::_device[] = {
+    SPIDesc("mpu9250",    0, 0, SPI_MODE_3, 8, SPI_CS_KERNEL,  1*MHZ, 11*MHZ),
+    SPIDesc("bmp280",     0, 1, SPI_MODE_3, 8, SPI_CS_KERNEL,  10*MHZ,10*MHZ),
+};
 #else
 // empty device table
 SPIDesc SPIDeviceManager::_device[] = {
